AudiusProject/audius-protocol
# New "soft unhealthy" env var + actually set healthy status from content node

### Description

This change actually sets the healthy status in the health_check response. That actually affects a few things:
- Node will no longer be selected during user signup
- CNodeHealthManager will consider the node unhealthy and start rolling users off
- RolloverNodes in libs sanity check will also rollover so we should merge and release  https://github.com/AudiusProject/audius-protocol/pull/3658 before we enable this.

This does not affect the node's ability to serve content or stay healthy, so in essence it will be serving requests for streaming content and older user uploads until state machine reconfigs the users off so there's no write block that will error for users trying to upload content.

### Tests

Added two unit tests to make sure health check returns healthy: false when env var is set and creator node selection respects this property.

To test the actual state machine flow I:
1. create a few users using `A seed create-user`
2. find one of the nodes that's both a secondary and primary and special case health check to return healthy: false
3. observe reconfigs happen to move off the unhealthy node

Logs
It correctly identifies the nodes as unhealthy and triggers an update-replica-set call.
```
[2022-08-24T22:35:12.777Z]  INFO: audius_creator_node/26871 on 325a80f8103c: New job: {"id":"1","name":"__default__","data":{"enqueuedBy":"find-replica-set-updates-queue#1","wallet":"0x9761511f1e0dcac340a75bb302767bc6961cdc71","userId":2,"primary":"http://cn1_creator-node_1:4000","secondary1":"http://cn3_creator-node_1:4002","secondary2":"http://cn2_creator-node_1:4001","unhealthyReplicas":["http://cn3_creator-node_1:4002"],"replicaToUserInfoMap":{"http://cn1_creator-node_1:4000":{"clock":2,"filesHash":"9adbbac18168c7d61f9c5d18d15bfa58"},"http://cn3_creator-node_1:4002":{"clock":2,"filesHash":"9adbbac18168c7d61f9c5d18d15bfa58"},"http://cn2_creator-node_1:4001":{"clock":2,"filesHash":"9adbbac18168c7d61f9c5d18d15bfa58"}},"enabledReconfigModes":["RECONFIG_DISABLED","ONE_SECONDARY","MULTIPLE_SECONDARIES","PRIMARY_AND_OR_SECONDARIES"]},"opts":{"removeOnComplete":100000,"removeOnFail":100000,"attempts":1,"delay":0,"timestamp":1661380512775},"progress":0,"delay":0,"timestamp":1661380512775,"attemptsMade":0,"stacktrace":[],"returnvalue":null,"finishedOn":null,"processedOn":1661380512776} (queue=update-replica-set-queue, jobId=1, logLevel=info)
[2022-08-24T22:35:12.778Z]  INFO: audius_creator_node/26871 on 325a80f8103c: New job: {"id":"2","name":"__default__","data":{"enqueuedBy":"find-replica-set-updates-queue#1","wallet":"0x944c559e8bef7f912777a63a66520b02054d82a9","userId":3,"primary":"http://cn1_creator-node_1:4000","secondary1":"http://cn3_creator-node_1:4002","secondary2":"http://cn2_creator-node_1:4001","unhealthyReplicas":["http://cn3_creator-node_1:4002"],"replicaToUserInfoMap":{"http://cn1_creator-node_1:4000":{"clock":2,"filesHash":"8835669133c7fd799648c942c2634c0d"},"http://cn3_creator-node_1:4002":{"clock":2,"filesHash":"8835669133c7fd799648c942c2634c0d"},"http://cn2_creator-node_1:4001":{"clock":2,"filesHash":"8835669133c7fd799648c942c2634c0d"}},"enabledReconfigModes":["RECONFIG_DISABLED","ONE_SECONDARY","MULTIPLE_SECONDARIES","PRIMARY_AND_OR_SECONDARIES"]},"opts":{"removeOnComplete":100000,"removeOnFail":100000,"attempts":1,"delay":0,"timestamp":1661380512775},"progress":0,"delay":0,"timestamp":1661380512775,"attemptsMade":0,"stacktrace":[],"returnvalue":null,"finishedOn":null,"processedOn":1661380512776} (queue=update-replica-set-queue, jobId=2, logLevel=info)
```

Also visible on bull dashboard

<img width="1470" alt="Bull_Dashboard" src="https://user-images.githubusercontent.com/295938/186536320-e727acca-7f9d-4659-9519-07532e95036c.png">


### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?

No new users on the node and users with the node on the replica set should go to zero.
